### PR TITLE
update go-multiaddr to v0.11.0

### PR DESCRIPTION
**Pull request by marten-seemann**
This is getting in two important changes in go-multiaddr that many users have been complaining about (see release notes).

### CODE REVIEW

File: p2p/protocol/identify/obsaddr.go [hunk 1]
```diff
@@ -214,11 +214,11 @@ func (oas *ObservedAddrManager) filter(observedAddrs []*observedAddr) []ma.Multi
...
 		// For ties, we prefer the ones with more votes.
-		slices.SortFunc(s, func(first, second *observedAddr) bool {
-			if first.numInbound > second.numInbound {
-				return true
+		slices.SortFunc(s, func(first, second *observedAddr) int {
+			if first.numInbound > second.numInbound || len(first.seenBy) > len(second.seenBy) {
+				return -1
 			}
-			return len(first.seenBy) > len(second.seenBy)
+			return 1
```
Comment (marten-seemann) [inline p2p/protocol/identify/obsaddr.go:221]: Please check the logic carefully.

File: p2p/protocol/identify/obsaddr.go [hunk 2]
```diff
@@ -214,11 +214,11 @@ func (oas *ObservedAddrManager) filter(observedAddrs []*observedAddr) []ma.Multi
 
 		// We prefer inbound connection observations over outbound.
 		// For ties, we prefer the ones with more votes.
-		slices.SortFunc(s, func(first, second *observedAddr) bool {
-			if first.numInbound > second.numInbound {
-				return true
+		slices.SortFunc(s, func(first, second *observedAddr) int {
+			if first.numInbound > second.numInbound || len(first.seenBy) > len(second.seenBy) {
```
Comment (MarcoPolo) [inline p2p/protocol/identify/obsaddr.go:218]: Old behavior is still here, and the new behavior seems reasonable. Should we update the comment?

**Review by MarcoPolo — APPROVED** [review]

File: p2p/protocol/identify/obsaddr.go (hunk 2 above)
Comment (marten-seemann) [reply p2p/protocol/identify/obsaddr.go:218]: > Old behavior is still here, and the new behavior seems reasonable.

What do you mean? I didn't mean to change the behavior. Did I mess up the logic?

File: p2p/protocol/identify/obsaddr.go (hunk 1 above)
Comment (sukunrt) [reply p2p/protocol/identify/obsaddr.go:221]: lgtm.

**Review by sukunrt — APPROVED** [review]
The new code is strictly equivalent.

**Comment by fasmat** [comment]
Will this change be included in `v0.29.3`? At the moment `github.com/multiformats/go-multiaddr` v0.11.0 prevents us from updating to `v0.29.2`.

**Comment by marten-seemann** [comment]
That’s not possible according to semver rules.

**Comment by fasmat** [comment]
Then I'll guess we have to wait for `v0.30.0` 🙂

**Pull request merged**